src/packet_sources/macos.rs

```diff
@@ -0,0 +1,229 @@
+use crate::messages::{IpPacket, NetworkCommand, NetworkEvent, TunnelInfo};
+use crate::network::MAX_PACKET_SIZE;
+use crate::packet_sources::ipc::from_redirector::Message::Packet;
+use crate::packet_sources::ipc::{FromRedirector, PacketWithMeta};
+use crate::packet_sources::{ipc, PacketSourceConf, PacketSourceTask};
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use home::home_dir;
+use nix::{sys::stat::Mode, unistd::mkfifo};
+use prost::Message;
+use std::io::Cursor;
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use tokio::net::unix::pipe;
+use tokio::sync::broadcast;
+use tokio::sync::mpsc::Sender;
+use tokio::sync::mpsc::{unbounded_channel, Receiver, UnboundedReceiver, UnboundedSender};
+
+pub const IPC_BUF_SIZE: usize = MAX_PACKET_SIZE + 4;
+
+#[allow(dead_code)]
+pub struct PipeServer {
+    ip_rx: pipe::Receiver,
+    ip_tx: pipe::Sender,
+    net_rx: pipe::Receiver,
+    net_tx: pipe::Sender,
+    filter_rx: pipe::Receiver,
+    filter_tx: pipe::Sender,
+    ip_path: PathBuf,
+    net_path: PathBuf,
+    filter_path: PathBuf,
+}
+
+impl PipeServer {
+    pub async fn new(ip_pipe: &str, net_pipe: &str, filter_pipe: &str) -> Result<Self> {
+        let home_dir = match home_dir() {
+            Some(ip_path) => ip_path,
+            None => Err(anyhow!("Failed to get home directory"))?,
+        };
+
+        let ip_path = Path::new(&home_dir).join(format!("Downloads/{}.pipe", &ip_pipe));
+        let net_path = Path::new(&home_dir).join(format!("Downloads/{}.pipe", &net_pipe));
+        let filter_path = Path::new(&home_dir).join(format!("Downloads/{}.pipe", &filter_pipe));
```

Comment:
Only Downloads or Movies or Pictures dir are avalaible to be written or read from swift (for now)